apps/src/lib/wasm_loader/mod.rs

```diff
@@ -260,67 +261,36 @@ pub async fn pre_fetch_wasm(wasm_directory: impl AsRef<Path>) {
 pub fn read_wasm(
     wasm_directory: impl AsRef<Path>,
     file_path: impl AsRef<Path>,
-) -> Vec<u8> {
+) -> eyre::Result<Vec<u8>> {
     // load json with wasm hashes
     let checksums = Checksums::read_checksums(&wasm_directory);
 
     if let Some(os_name) = file_path.as_ref().file_name() {
         if let Some(name) = os_name.to_str() {
-            match checksums.0.get(name) {
+            let wasm_path = match checksums.0.get(name) {
                 Some(wasm_filename) => {
-                    let wasm_path = wasm_directory.as_ref().join(wasm_filename);
-                    match fs::read(&wasm_path) {
-                        Ok(bytes) => {
-                            return bytes;
-                        }
-                        Err(_) => {
-                            eprintln!(
-                                "File {} not found. ",
-                                wasm_path.to_string_lossy()
-                            );
-                            safe_exit(1);
-                        }
-                    }
+                    wasm_directory.as_ref().join(wasm_filename)
                 }
                 None => {
                     if !file_path.as_ref().is_absolute() {
-                        match fs::read(
-                            wasm_directory.as_ref().join(file_path.as_ref()),
-                        ) {
-                            Ok(bytes) => {
-                                return bytes;
-                            }
-                            Err(_) => {
-                                eprintln!(
-                                    "Could not read file {}. ",
-                                    file_path.as_ref().to_string_lossy()
-                                );
-                                safe_exit(1);
```

Comment:
I saw safe_exit does some flushing of stdout/stderr, in what contexts do we need this? I think any flushing could be better handled up the stack where necessary (e.g. in the case of running tests we shouldn't need it)

Reply:
Yeah it'd be good to consolidate our calls to `safe_exit` to somewhere further up the stack where it's less surprising, at some point could take a look at this across our codebase